src/tsickle.ts

```diff
@@ -367,12 +367,14 @@ class ClosureRewriter extends Rewriter {
     // We need to search backwards for the first JSDoc comment to avoid ignoring such when another
     // code-level comment is between that comment and the function declaration (see
     // testfiles/doc_params for an example).
-    let startPosition = 0;
+    let fullStartPosition = 0;
     let parsed: jsdoc.ParsedJSDocComment|null = null;
     for (let i = comments.length - 1; i >= 0; i--) {
       const {pos, end} = comments[i];
-
-      if (end <= this.file.getStart() && this.file.text.substring(end).startsWith('\n\n')) {
+      // end is relative within node.getFullText(), add getFullStart to obtain coordinates that are
+      // comparable to node positions.
+      const fullEnd = end + node.getFullStart();
```

Comment:
It's weird that you have both "fullEnd" and "end" in different coordinate systems, while "start" vs "fullStart" on a node are two positions in the same coordinate system.

Maybe you mean a different word than "full" here?  Like perhaps "fileEnd" or some such?  Or maybe rename "end" to "len" or something?

Same with renaming startPosition to fullStartPosition, it seems like the wrong name.  "full" means whether it includes trivia or not.

Reply:
Good point. I changed both to `docRelative*` - does that help?